actionmailbox/config/routes.rb

```diff
@@ -6,6 +6,16 @@
     post "/relay/inbound_emails"    => "relay/inbound_emails#create",    as: :rails_relay_inbound_emails
     post "/sendgrid/inbound_emails" => "sendgrid/inbound_emails#create", as: :rails_sendgrid_inbound_emails
 
+    # Amazon requires that SNS topic subscriptions have been accepted before sending notifications.
+    post "/amazon/inbound_emails"   => "amazon/confirmations#create", as: :rails_amazon_confirmations,
+      constraints: lambda { |request|
+        ActionMailbox::Ingresses::Amazon::SnsMessageTypeConstraint.new("SubscriptionConfirmation").matches?(request)
+      }
+    post "/amazon/inbound_emails"   => "amazon/inbound_emails#create",    as: :rails_amazon_inbound_emails,
+      constraints: lambda { |request|
+        ActionMailbox::Ingresses::Amazon::SnsMessageTypeConstraint.new("Notification").matches?(request)
+      }
+
```

Comment:
Is a 404 Not Found an appropriate response for other kinds of SNS messages? Are other SNS messages expected?